README.md

```diff
@@ -1 +1,152 @@
+
+```javascript
+// data/local.js
+module.exports = {
+  baseUrl: 'http://localhost:3000'
+});
+```
+
+You can also add baseUrl in the configuration of the sitemap module. You may not like to modify or overwrite the `baseUrl` for the site or prefer to not use a `baseUrl` for other purposes.
+
+```javascript
+// app.js
+require('apostrophe')({
+  shortName: 'my-project',
+  modules: {
+    '@apostrophecms/sitemap': {
+      baseUrl: 'https://example.com'
+    }
+  }
+});
+```
+
+### Options
+
+All sitemap module options are configured in an `options` object.
+
+```javascript
+// modules/@apostrophecms/sitemap/index.js
+module.exports = {
+  // 👇 Module options
+  options: {
+    cacheLifetime: 1800,
+    excludeTypes: [ 'exclusive-page', 'category' ]
+    piecesPerBatch: 500
+  }
+};
+```
+
+These can be added in the `app.js` configuration object for the module, but it is better practice use a dedicated file for module configuration.
+
+#### `cacheLifetime`
+
+By default sitemaps are cached for one hour. You can change this by specifying the `cacheLifetime` option to this module, in seconds.
+
+**Tip:** To make entering the cache lifetime easier it can help to write it as a math expression, multiplying the desired number of minutes by sixty:
+
+```javascript
+cacheLifetime: 30 * 60 // or 1800 seconds
+```
+
+Keep in mind: Google and other search engines more than weekly, if that.Refreshing once every hour is usually more than often enough.
+
+#### `excludeTypes`
+
+If there are particular page types or piece content types that should *not* be in the sitemap, list them in an array as the `excludeType` option.
+
+```javascript
+excludeTypes: [ 'exclusive-page', 'category' ]
+```
+
+#### `piecesPerBatch`
+
+If you have thousands of URLs to index, building the sitemap may take a long time. By default, this module processes 100 pieces at a time, to avoid using too much memory. You can adjust this by setting the `piecesPerBatch` option to a larger number. Be aware that if you have many fields and content relationships **it is possible this can use a great deal of memory**.
+
+```javascript
+piecesPerBatch: 500
+```
+
+### Tasks
+
+#### `map`
+
+The `map` command with generate an up-to-date sitemap on demand. The base command **prints the sitemap into the console**, or allows you to pipe it as needed, to help generate a static file version.On the command line, run:
+
+```bash
+node app @apostrophecms/sitemap:map
+```
+
+Use the `--update-cache` option to force a cache update at any time. The map will not print to the console with this option. If the website is very large (multiple hundreds of URLs), running this task option with a cron job on the production server more often than the standard cache refresh can help ensure the cache is available when a search engine begins crawling the site.
```

Comment:
Needing a secondary flag to get `/sitemap.xml` to update puts `sitemap.xml` behind the console print out in terms of priority, which doesn't feel like it makes sense for a web application.

If anything, a console print only situation should be the thing behind a flag

Reply:
@boutell This was part of the A2 API. Referring to the example in the README (also from A2) do you know why it'd be better to run this command twice as often as the lifetime rather than simply halve the cache lifetime?

Reply:
Never you mind. I'm splitting the task up.

Reply:
The idea here is to avoid the cache ever expiring. If the cache lifetime is 30 minutes and the cron job is every 30 minutes, it can expire while waiting for that cron job, and we still have a pileup of requests while it's not ready.

Reply:
So it might be better if there was a way to disable the cache lifetime altogether so it just refreshes as often as your cron job runs and we don't worry about ever trying to do it in direct response to a request when that feature is in use.

(For smaller sites, doing it in direct response is fine and saves hassling with cron.)